src/lib/rest/httpRequestSend.cpp

```diff
@@ -239,6 +239,9 @@ static int contentLenParse(char* s)
 */
 int httpRequestSend
 (
+   int                                        providerLimit,
+   int                                        providerOffset,
+   ApiVersion                                 apiVersion,
    CURL*                                      _curl,
```

Comment:
As a common practise, we keep the existing argument order in functions and add new arguments at the end (i.e. after timeoutInMilliseconds).

Reply:
Moreover, I think `apiVersion` can be removed.

Reply:
I think, new parameters should be added before the `timeoutInMilliseconds`, as `timeoutInMilliseconds` is a default parameter.

So if we are keeping the newly added parameters after the `timeoutInMilliseconds` then it is taking incorrect values in `httpRequestSend.cpp` file, (I have added LM_E level log to print the values of new parametes), for e.g.


**If we keep the parameters after the `timeoutInMilliseconds`:**
Actual value of new parameters inside `postQueryContext.cpp` file, before `httpRequestSend.cpp()` function call: 
<pre>
time=2023-05-19T04:06:05.984Z | lvl=ERROR | corr=7a3c718e-f5fa-11ed-8540-e3fa43514f13 | trans=1684469149-080-00000000001 | from=127.0.0.1 | srv=<none> | subsrv=<none> | comp=Orion | op=postQueryContext.cpp[305]:queryForward | msg=Arqam-17may23 Error (providerLimit value in postQueryContext/queryForward at L304: '15')
time=2023-05-19T04:06:05.984Z | lvl=ERROR | corr=7a3c718e-f5fa-11ed-8540-e3fa43514f13 | trans=1684469149-080-00000000001 | from=127.0.0.1 | srv=<none> | subsrv=<none> | comp=Orion | op=postQueryContext.cpp[306]:queryForward | msg=Arqam-17may23 Error (providerOffset value in postQueryContext/queryForward at L305: '0')
</pre>

Incorrect value of new parameters being passed inside `httpRequestSend.cpp` file:
<pre>
time=2023-05-19T04:06:05.984Z | lvl=ERROR | corr=7a3c718e-f5fa-11ed-8540-e3fa43514f13 | trans=1684469149-080-00000000001 | from=127.0.0.1 | srv=<none> | subsrv=<none> | comp=Orion | op=httpRequestSend.cpp[267]:httpRequestSend | msg=Arqam-17may23 Error (providerLimit value in httpRequestSend at L267: '0')
time=2023-05-19T04:06:05.984Z | lvl=ERROR | corr=7a3c718e-f5fa-11ed-8540-e3fa43514f13 | trans=1684469149-080-00000000001 | from=127.0.0.1 | srv=<none> | subsrv=<none> | comp=Orion | op=httpRequestSend.cpp[268]:httpRequestSend | msg=Arqam-17may23 Error (providerOffset value in httpRequestSend at L268: '0')
</pre>

**If we keep the parameters before the parameter `timeoutInMilliseconds`:**
In that case correct values are being received in `httpRequestSend.cpp` file (I have added LM_E level log to print the values):


Actual values of new parameters inside `postQueryContext.cpp` file, before `httpRequestSend.cpp()` function call: 
```
time=2023-05-19T04:16:58.653Z | lvl=ERROR | corr=ff41bbfe-f5fb-11ed-8540-e3fa43514f13 | trans=1684469739-808-00000000001 | from=127.0.0.1 | srv=<none> | subsrv=<none> | comp=Orion | op=postQueryContext.cpp[305]:queryForward | msg=Arqam-17may23 Error (providerLimit value in postQueryContext/queryForward at L304: '15')
time=2023-05-19T04:16:58.653Z | lvl=ERROR | corr=ff41bbfe-f5fb-11ed-8540-e3fa43514f13 | trans=1684469739-808-00000000001 | from=127.0.0.1 | srv=<none> | subsrv=<none> | comp=Orion | op=postQueryContext.cpp[306]:queryForward | msg=Arqam-17may23 Error (providerOffset value in postQueryContext/queryForward at L305: '0')
```
Correct values of new parameters being passed inside `httpRequestSend.cpp` file:
```
time=2023-05-19T04:16:58.653Z | lvl=ERROR | corr=ff41bbfe-f5fb-11ed-8540-e3fa43514f13 | trans=1684469739-808-00000000001 | from=127.0.0.1 | srv=<none> | subsrv=<none> | comp=Orion | op=httpRequestSend.cpp[267]:httpRequestSend | msg=Arqam-17may23 Error (providerLimit value in httpRequestSend at L267: '15')
time=2023-05-19T04:16:58.653Z | lvl=ERROR | corr=ff41bbfe-f5fb-11ed-8540-e3fa43514f13 | trans=1684469739-808-00000000001 | from=127.0.0.1 | srv=<none> | subsrv=<none> | comp=Orion | op=httpRequestSend.cpp[268]:httpRequestSend | msg=Arqam-17may23 Error (providerOffset value in httpRequestSend at L268: '0')
```

Reply:
Suggested signature:

```
extern int httpRequestSend
(
  CURL*                                      curl,
  const std::string&                         idStringForLogs,
  ...
  long long*                                 statusCodeP,
  const std::map<std::string, std::string>&  extraHeaders,
  int                                        providerLimit = -1,
  int                                        providerOffset = -1,
  const std::string&                         acceptFormat          = "",
  long                                       timeoutInMilliseconds = -1
);
```

Note the suggestion to make providerLimit and priverOffset default parameter. The usage of -1 as default is aligned with previous comments and would make some parts of the code simpler (the ones in notification logic and in update forward logic, where CPrs pagination doesn't make sense).

Reply:
Fixed in [1e47b07](https://github.com/telefonicaid/fiware-orion/pull/4275/commits/1e47b077ce86f15d09438331b764710751bf1910)